internal/fake/factory.go

```diff
@@ -81,7 +81,7 @@ func (f *Factory) Manager() (target.Manager, error) {
 	return target.NewManager(f.Config, f.TargetProviderImpl, f.ClientProviderImpl, f.KubeconfigCacheImpl)
 }
 
-func (f *Factory) GetConfigFile() string {
+func (f *Factory) ConfigFile() string {
```

Comment:
We don't need to extend the factory with this method. The only place where we need the filename of the gardenctl configuration is the Config object. In our case it is a struct. We could extend this struct :
```go
// Config holds the gardenctl configuration
type Config struct {
	Filename string `yaml:"-"`
	Gardens []Garden `yaml:"gardens"`
	MatchPatterns []string `yaml:"matchPatterns"`
}
```
Thats it!